node/src/types/deploy.rs

```diff
@@ -827,27 +828,380 @@ impl Deploy {
+
+        Deploy::new(
+            self.header().timestamp(),
+            self.header().ttl(),
+            self.header().gas_price(),
+            self.header().dependencies().clone(),
+            self.header().chain_name().to_string(),
+            self.payment().clone(),
+            session,
+            &secret_key,
+            None,
+        )
+    }
+
+    #[cfg(test)]
+    pub(crate) fn with_missing_entry_point_in_session_contract(
+        &mut self,
+        rng: &mut TestRng,
+    ) -> Self {
+        let session = ExecutableDeployItem::StoredContractByName {
+            name: "Test".to_string(),
+            entry_point: "non-existent-entry-point".to_string(),
+            args: Default::default(),
+        };
+        let secret_key = SecretKey::random(rng);
+
+        Deploy::new(
+            self.header().timestamp(),
+            self.header().ttl(),
+            self.header().gas_price(),
+            self.header().dependencies().clone(),
+            self.header().chain_name().to_string(),
+            self.payment().clone(),
+            session,
+            &secret_key,
+            None,
+        )
+    }
+
+    #[cfg(test)]
+    pub(crate) fn with_valid_session_package_by_name(&mut self, rng: &mut TestRng) -> Self {
+        let session = ExecutableDeployItem::StoredVersionedContractByName {
+            name: "Test".to_string(),
+            version: None,
+            entry_point: "call".to_string(),
+            args: Default::default(),
+        };
+
+        let secret_key = SecretKey::random(rng);
+
+        Deploy::new(
+            self.header().timestamp(),
+            self.header().ttl(),
+            self.header().gas_price(),
+            self.header().dependencies().clone(),
+            self.header().chain_name().to_string(),
+            self.payment().clone(),
+            session,
+            &secret_key,
+            None,
+        )
+    }
+
+    #[cfg(test)]
+    pub(crate) fn with_missing_session_package_by_hash(&mut self, rng: &mut TestRng) -> Self {
+        let session = ExecutableDeployItem::StoredVersionedContractByHash {
+            hash: Default::default(),
+            version: None,
+            entry_point: "call".to_string(),
+            args: Default::default(),
+        };
+        let secret_key = SecretKey::random(rng);
+
+        Deploy::new(
+            self.header().timestamp(),
+            self.header().ttl(),
+            self.header().gas_price(),
+            self.header().dependencies().clone(),
+            self.header().chain_name().to_string(),
+            self.payment().clone(),
+            session,
+            &secret_key,
+            None,
+        )
+    }
+
+    #[cfg(test)]
+    pub(crate) fn with_missing_contract_version_in_session_package(
+        &mut self,
+        rng: &mut TestRng,
+    ) -> Self {
+        let session = ExecutableDeployItem::StoredVersionedContractByName {
+            name: "Test".to_string(),
+            version: Some(6u32),
```

Comment:
`version` in `session` is `Some`, yet the function name says `with_missing_contract_version_in_session...`, maybe it's just a naming issue.

Reply:
Naming issue, yes, its meant to be Some version that doesn't exist in a contract package

Reply:
Renamed the tests around it